namer/namer.cc

```diff
@@ -452,23 +452,45 @@ class NameInserter {
     }
 
     unique_ptr<ast::Expression> postTransformSend(core::MutableContext ctx, unique_ptr<ast::Send> original) {
-        ast::MethodDef *mdef;
-        if (original->args.size() == 1 && (mdef = ast::cast_tree<ast::MethodDef>(original->args[0].get())) != nullptr) {
+        core::SymbolRef method;
+        if (original->args.size() == 1) {
+            if (auto mdef = ast::cast_tree<ast::MethodDef>(original->args[0].get())) {
```

Comment:
There's a fair number of opportunities to bail out early here, but we're also only going to care about this logic if the send is in fact one of the names checked below (`private` or `protected` or what-have-you) which makes me wonder if we should guard against that case first, and always bail early if we're not looking at a method like `private`. It might only save a small amount of time, but I'm also extra-sensitive about taking more time than necessary in the namer.